third_party/terraform/resources/resource_composer_environment.go.erb

```diff
@@ -198,6 +238,28 @@ func resourceComposerEnvironment() *schema.Resource {
 								},
 							},
 						},
+						"private_environment_config": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Computed: true,
+							MaxItems: 1,
+							ForceNew: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"enable_private_endpoint": {
+										Type:     schema.TypeBool,
+										Required: true,
```

Comment:
It doesn't make sense to have all optional fields in an optional block, does it? That would allow magic like:

```
...
private_environment_config {}
...
```

Reply:
we'd set it to optional/default so if it's not specified, the default is used. Default and required can't be used together since required just determines whether the field must be given in config, so you couldn't default to the field.

Reply:
Gotcha.